### DISCUSSION TITLE
Is it possible to replace Javascript with Stylus?

### DISCUSSION THREAD

**Discussion by Zerophase** [openstyles/stylus#1652] [Q&A]
I have a few websites, which I want to modify the behavior of a Javascript function. Is that possible with Stylus? Have an example?

**Comment by tophf** [comment]
No, Stylus is for CSS only.

**Comment by dmland** [comment]
Zerophase,

It would depend entirely on the JavaScript function in question, so in general, the answer is no: JavaScript  can do many things that CSS simply can't.

On the other hand, I've seen some darn amazing interactive experiences constructed entirely in CSS, but those are generally exercises in helping developers understand the furthest corners of CSS, rather than replacements for JavaScript.

I imagine that clever people here could probably cook up any number of contrived examples of emulating "javascript-like behavior" using pure CSS, but without knowing the specifics, it's more likely not possible.

If I'm wrong, I'm confident that someone will put me right.

Dave
    On Saturday, September 2, 2023 at 10:18:52 AM PDT, Zerophase ***@***.***> wrote:  
 
 


I have a few websites, which I want to modify the behavior of a Javascript function. Is that possible with Stylus? Have an example?

**Reply by Zerophase** [reply]
Oh yeah, CSS can do a lot of what people use Javascript for. Probably runs faster too.
